dbt/parser/schemas.py

```diff
@@ -61,186 +62,386 @@ def as_kwarg(key, value):
+                      description: ${column_2_description}
+                      tests:
+                          - ${constraint_type}: {$keyword_args_dict}
+                          ...
+                - name: ${other_model_name}
+                  ...
+        """
+        if 'models' not in test_yml:
+            # You could legitimately not have any models in your schema.yml, if
+            # sources were supported
+            return
+
+        for model in test_yml['models']:
+            try:
+                model = UnparsedNodeUpdate(**model)
+            except dbt.exceptions.ValidationException as exc:
+                # we don't want to fail the full run, but we do want to fail
+                # parsing this file
+                msg = "Invalid test config given in {} near {}: {}".format(
+                        original_file_path, model, exc
+                )
+                if dbt.flags.STRICT_MODE:
+                    dbt.exceptions.raise_compiler_error(msg, model)
+
+                dbt.utils.compiler_warning(model.get('name'), msg)
+                continue
 
-                    for config in configs:
-                        to_add = cls.get_parsed_schema_test(
-                                    test, test_type, model_name, config,
-                                    root_project, projects, macros)
+            iterator = cls.parse_model(model, package_name, root_dir,
+                                       original_file_path, root_project,
+                                       all_projects, macros)
 
-                        if to_add is not None:
-                            to_return[to_add.get('unique_id')] = to_add
+            for node_type, node in iterator:
+                yield node_type, node
 
-        return to_return
+    @classmethod
+    def parse_model(cls, model, package_name, root_dir, path, root_project,
+                    all_projects, macros):
+        """Given an UnparsedNodeUpdate, return column info about the model
+
+            - column info (name and maybe description) as a dict
+            - a list of ParsedNodes repreenting tests
+
+        This is only used in parsing the v2 schema.
+        """
+        model_name = model['name']
+
+        column_info = []
+        for column in model.get('columns', []):
+            column_name = column['name']
+            column_info.append({
+                'name': column_name,
+                'description': column.get('description', '')
+            })
+            for test in column.get('tests', []):
+                test_type, test_args = cls._build_v2_test_args(
+                    test, column_name
+                )
+                node = cls.build_node(
+                    model_name, package_name, test_type, test_args, root_dir,
+                    path, root_project, all_projects, macros
+                )
+                yield 'test', node
+
+        for test in model.get('tests', []):
```

Comment:
I commented above about support for testing expressions, having forgotten about this part of the spec. I think this could totally work -- ignore the comment above :)